pmd-java/src/test/resources/net/sourceforge/pmd/lang/java/rule/codestyle/xml/UnnecessaryImport.xml

```diff
@@ -1474,8 +1474,281 @@ public class Foo {
+        <code><![CDATA[
+import some.pkg.MyEnum;
+@MyAnnotation({MyEnum.VALUE1, MyEnum.VALUE2})
+public class Foo {}
+        ]]></code>
+    </test-code>
+
+    <!-- Test cases for imports used in lambda parameters -->
+    <test-code>
+        <description>Import used in lambda parameter type</description>
+        <expected-problems>0</expected-problems>
+        <code><![CDATA[
+import java.io.File;
+public class Foo {
+    void test() {
+        list.stream().map((File f) -> f.getName());
+    }
+}
+        ]]></code>
+    </test-code>
+
+    <!-- Test cases for imports used in method references -->
+    <test-code>
+        <description>Import used in method reference</description>
+        <expected-problems>0</expected-problems>
+        <code><![CDATA[
+import java.io.File;
+public class Foo {
+    void test() {
+        list.stream().map(File::getName);
+    }
+}
+        ]]></code>
+    </test-code>
+
+    <!-- Test cases for imports used in switch expressions -->
+    <test-code>
+        <description>Import used in switch expression</description>
+        <expected-problems>0</expected-problems>
+        <code><![CDATA[
+import some.pkg.MyEnum;
+public class Foo {
+    String test(MyEnum e) {
+        return switch(e) {
+            case VALUE1 -> "one";
+            case VALUE2 -> "two";
+        };
+    }
+}
+        ]]></code>
+    </test-code>
+
+    <!-- Test cases for imports used in records -->
+    <test-code>
+        <description>Import used in record component</description>
+        <expected-problems>0</expected-problems>
+        <code><![CDATA[
+import java.time.LocalDate;
+public record Person(String name, LocalDate birthDate) {}
+        ]]></code>
+    </test-code>
+
+    <!-- Test cases for imports used in sealed classes -->
+    <test-code>
+        <description>Import used in sealed class hierarchy</description>
+        <expected-problems>0</expected-problems>
+        <code><![CDATA[
+import some.pkg.Shape;
+public sealed class Circle extends Shape permits BigCircle, SmallCircle {}
+        ]]></code>
+    </test-code>
+
+    <!-- Test cases for imports used in pattern matching -->
+    <test-code>
+        <description>Import used in instanceof pattern matching</description>
+        <expected-problems>0</expected-problems>
+        <code><![CDATA[
+import java.util.List;
+public class Foo {
+    void test(Object o) {
+        if (o instanceof List l) {
+            l.size();
+        }
+    }
+}
+        ]]></code>
+    </test-code>
+
+    <!-- Test cases for imports used in text blocks -->
+    <test-code>
+        <description>Import used in text block with embedded expressions</description>
+        <expected-problems>0</expected-problems>
+        <code><![CDATA[
+import java.time.LocalDate;
+public class Foo {
+    String test(String name) {
+        return """
+            Hello %s!
+            Today is %s
+            """.formatted(name, LocalDate.now());
+    }
 }
```

Comment:
Mabye this test case could be simplified? Text block is a string and text block is irrelevant here.

You can achieve the same effect with that:

```java
import java.time.LocalDate;
public class Foo {
  {
     LocalDate.now();
  }
}
```